var/spack/repos/builtin/packages/occa/package.py

```diff
@@ -33,91 +38,177 @@ class Occa(Package):
+            env.set("OCCA_INCLUDE_PATH", hip_dir.include)
+            env.set("OCCA_LIBRARY_PATH", ":".join(hip_libs.directories))
+            env.set("OCCA_HIP_ENABLED", "1")
+        else:
+            env.set("OCCA_HIP_ENABLED", "0")
 
-        if "~opencl" in spec:
-            env.set("OCCA_OPENCL_ENABLED", "0")
+        # Turn-off OpenCL since it fails on some of the machines.
+        env.set("OCCA_OPENCL_ENABLED", "0")
 
-        if "~openmp" in spec:
+        if "+openmp" in spec:
+            env.set("OCCA_OPENMP_ENABLED", "1")
+        else:
             env.set("OCCA_OPENMP_ENABLED", "0")
 
         # Setup run-time environment for testing.
         env.set("OCCA_VERBOSE", "1")
-        self._setup_runtime_flags(env)
-
-    def setup_run_environment(self, env):
-        # The 'env' is included in the Spack generated module files.
-        self._setup_runtime_flags(env)
+        self.setup_run_environment(env)
 
     def setup_dependent_build_environment(self, env, dependent_spec):
         # Export OCCA_* variables for everyone using this package from within
         # Spack.
-        self._setup_runtime_flags(env)
+        self.setup_run_environment(env)
+
+
+class CMakeBuilder(CMakeBuilder, SetupEnvironment):
+    def cmake_args(self):
+        spec = self.spec
+        args, include_dirs, library_dirs = [], [], []
+
+        if "+cuda" in spec:
+            cuda_dir = spec["cuda"].prefix
+            cuda_libs_list = ["libcuda", "libcudart", "libOpenCL"]
+            cuda_libs = find_libraries(cuda_libs_list, cuda_dir, shared=True, recursive=True)
+            include_dirs.append(cuda_dir.include)
+            library_dirs.extend(cuda_libs.directories)
+            args.append("-DOCCA_ENABLE_CUDA=ON")
+        else:
+            args.append("-DOCCA_ENABLE_CUDA=OFF")
+
+        if "+rocm" in spec:
+            hip_dir = spec["hip"].prefix
+            hip_libs_list = ["amdhip64"]
+            hip_libs = find_libraries(hip_libs_list, hip_dir, shared=True, recursive=True)
+            include_dirs.append(hip_dir.include)
+            library_dirs.extend(hip_libs.directories)
+            args.append("-DOCCA_ENABLE_HIP=ON")
+        else:
+            args.append("-DOCCA_ENABLE_HIP=OFF")
+
+        # Turn-off OpenCL since it fails on some of the machines.
+        args.append("-DOCCA_ENABLE_OPENCL=OFF")
+
+        if "+openmp" in spec:
+            args.append("-DOCCA_ENABLE_OPENMP=ON")
+        else:
+            args.append("-DOCCA_ENABLE_OPENMP=OFF")
+
+        if "+debug" in spec:
+            args.append("-DCMAKE_BUILD_TYPE=Debug")
+
+        args.append("-DCMAKE_INCLUDE_PATH={0}".format(";".join(include_dirs)))
+        args.append("-DCMAKE_LIBRARY_PATH={0}".format(";".join(library_dirs)))
+
+        return args
+
+    @run_after("build")
+    @on_package_attributes(run_tests=True)
+    def check(self):
+        if self.pkg.run_tests:
+            make("-C", self.prefix, "test", parallel=False)
```

Comment:
:thinking: Is this needed because of `parallel=False`?

Reply:
It is not clear to me what `it` refers to. Is it the entire `make` command or a part of it?